### ISSUE TITLE
TS 2.0b / 2.0.2 RC crashes Visual Studio with no helpful errors if the Node.js is outdated.

### ISSUE THREAD

**Issue by wizzard0** [Microsoft/TypeScript#10983]
TLDR: after installing TS 2.0.2 RC Visual Studio stops opening TypeScript files unless systemwide Node is updated to 6.6.0
##### Expected Behavior

Message "Update node.js" is displayed, or the file opens.
##### Actual Behavior

Typescript language service crashes, WerFault.exe runs, VS hangs for 2 minutes, then restarts.

NOTE: initially tried updating VS Update 3 KB3165756, 2.0 beta / 2.0.2 RC / NTVS 1.2/ NTVS latest to no avail.

Node v5.7.0 - does not work, v6.6.0 - works

ActivityLog.xml:

```
  <entry>
    <record>723</record>
    <time>2016/09/19 10:16:55.250</time>
    <type>Error</type>
    <source>Editor or Editor Extension</source>
    <description>Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Hosting.JavaScriptException: Intl is not available.&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Hosting.JavaScriptEngine.ThrowIfError(JavaScriptErrorCode error)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Hosting.JavaScriptEngine..ctor(String text, String url, INativeEngine nativeEngine)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.ShimFactory.ShimFactory.Factory.ReinitializeEngineAndFactoryIfNeeded()&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.ShimFactory.ShimFactory.Factory.CreateOrUpdateShim(String createShimFunction, Object host, ShimAndServicesFileVersion&amp; shimInfo)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.ShimFactory.ShimFactory.CreateOrUpdateShim(String caption, String createShimFunction, Object host, ShimAndServicesFileVersion&amp; shimInfo)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Proxies.AbstractProxy`1.InvokeShimMethod[T](String methodName, Object[] arguments)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Proxies.ClassifierProxy.TryAddLexicalClassifications(String lineText, Int32 lexState, Int32 offset, TypeScriptLexicalClassificationMap classificationMap, ITextSnapshot snapshot, List`1 tagSpans, Int32&amp; newEndOfLineState)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.Features.Classifier.LexicalClassificationTaggerProvider.Tagger.AddLexicalClassifications(ClassifierProxy classifier, String text, Int32 lineNum, Int32 offset, ITextSnapshot snapshot, List`1 tagSpans)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.Features.Classifier.LexicalClassificationTaggerProvider.Tagger.GetClassifications(NormalizedSnapshotSpanCollection spans, List`1 result)&#x000D;&#x000A;   at Microsoft.CodeAnalysis.Editor.TypeScript.Features.Classifier.LexicalClassificationTaggerProvider.Tagger.GetTags(NormalizedSnapshotSpanCollection spans)&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.&lt;GetTagsForBuffer&gt;d__38.MoveNext()</description>
  </entry>
```

Microsoft Visual Studio Community 2015
Version 14.0.25431.01 Update 3
Microsoft .NET Framework
Version 4.6.02001

Installed Version: Community

Visual Basic 2015   00322-20000-00000-AA393
Microsoft Visual Basic 2015

Visual C# 2015   00322-20000-00000-AA393
Microsoft Visual C# 2015

Visual C++ 2015   00322-20000-00000-AA393
Microsoft Visual C++ 2015

Visual F# 2015   00322-20000-00000-AA393
Microsoft Visual F# 2015

Windows Phone SDK 8.0 - ENU   00322-20000-00000-AA393
Windows Phone SDK 8.0 - ENU

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11107.0
ASP.NET and Web Tools 2015.1 (Beta8)

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Clang with Microsoft CodeGen   14.0.25515
Clang with Microsoft CodeGen

Common Azure Tools   1.8
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

HideMenu   1.0
Hides the Visual Studio main menu, similar to Windows Explorer and Internet Explorer

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Node.js Tools   1.2.40915.01
Adds support for developing and debugging Node.js apps in Visual Studio

Node.js Tools - Profiling   1.2.40915.01
Profiling support for Node.js projects.

NuGet Package Manager   3.4.4
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

Text Sharp   2.0
Visual Studio IDE text clarity tuner. An easy way to turn off ClearType.

TypeScript   2.0.2.0
TypeScript tools for Visual Studio
##### Steps to Reproduce
1. Try to open TS file

**Comment by mhegazy** [comment]
The typescript language service does not use node any way. so i am not sure what the node instllation has to do with all of this. 

The error you added suggests that ChackraCore.dll is not installed, or could not be loaded for some reason.

can you share the version numbers for:
- C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TypeScript\chakracore.dll
- C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TypeScript\Microsoft.CodeAnalysis.TypeScript.EditorFeatures.dll

also can you try:
- uninstall all versions of TS from add/remove programs
- install latest update 3: https://msdn.microsoft.com/en-us/library/mt752379.aspx
- install TS 2.0.2 again.

also capture the install logs for this last part, you can find them in `%TEMP%`, and they should have a prefix of `dd_TypeScript*`

**Comment by wizzard0** [comment]
ChakraCore: 1.2.65535.65535
TypeScript.EditorFeatures: 14.30828.1.0

After updating Node.js it currently works fine, so I can't really reproduce the issue. Will definitely report more if there will be a standalone repro, maybe at weekend.

**Comment by mhegazy** [comment]
do you see any relevant events in the event log around the time of the crash?

**Comment by wizzard0** [comment]
Reproduced on another machine.
The problem is caused by TypeScript 1.7 and 1.8 SDK being present in program files / microsoft SDKs.
The bug goes away if I rename the folder, and returns if I bring it back.
Add-remove programs and repair does not help, deleting folders helps.

**Comment by dopare** [comment]
Getting exception after installing TS 2.0.3 on Visual Studio 2015

```
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Hosting.JavaScriptException: Object 
doesn&apos;t support this action&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Hosting.JavaScriptEngine.ThrowIfError(JavaScrip
tErrorCode error)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Hosting.JavaScriptEngine..ctor(String text, String 
url, INativeEngine nativeEngine)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.ShimFactory.ShimFactory.Factory.ReinitializeEngi
neAndFactoryIfNeeded()&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.ShimFactory.ShimFactory.Factory.CreateOrUpdat
eShim(String createShimFunction, Object host, ShimAndServicesFileVersion&amp; 
shimInfo)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.ShimFactory.ShimFactory.CreateOrUpdateShim(
String caption, String createShimFunction, Object host, ShimAndServicesFileVersion&amp; 
shimInfo)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Proxies.AbstractProxy`1.InvokeShimMethod[T]
(String methodName, Object[] arguments)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.Proxies.ClassifierProxy.TryAddLexicalClassificatio
ns(String lineText, Int32 lexState, Int32 offset, TypeScriptLexicalClassificationMap classificationMap, 
ITextSnapshot snapshot, List`1 tagSpans, Int32&amp; newEndOfLineState)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.Features.Classifier.LexicalClassificationTaggerProvider.Tagger.A
ddLexicalClassifications(ClassifierProxy classifier, String text, Int32 lineNum, Int32 offset, ITextSnapshot 
snapshot, List`1 tagSpans)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.Features.Classifier.LexicalClassificationTaggerProvider.Tagger.G
etClassifications(NormalizedSnapshotSpanCollection spans, List`1 result)&#x000D;&#x000A;   at 
Microsoft.CodeAnalysis.Editor.TypeScript.Features.Classifier.LexicalClassificationTaggerProvider.Tagger.G
etTags(NormalizedSnapshotSpanCollection spans)&#x000D;&#x000A;   at 
Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.&lt;GetTagsForBuffer&gt;d__38.MoveNext()
```

**Comment by mhegazy** [comment]
What version of windows are you running on?

**Comment by dopare** [comment]
Windows Server 2012

**Comment by mhegazy** [comment]
Looks like https://github.com/Microsoft/TypeScript/issues/11110; we are planning a release next week with the fix (2.0.6).

**Comment by wizzard0** [comment]
I experienced that on Windows 10 Pro (various builds), will test 2.0.6 once it will be released
